### PR TITLE
dotnet-*: update to 6.0.416;7.0.403 & add missing files

### PR DESCRIPTION
**Pull request by silverbzez**
#### Description

As promised, some file changes:

- `dotnet-cli`: not providing `host` folder any longer as `hostfxr`s should be shipped with matching runtimes
- `aspnetcore-runtime-2.1`: add missing `Microsoft.AspNetCore.All` folder that contains extra runtimes and plugins
- `aspnetcore-runtime-*`: add `dotnet-runtime-*` as dependency
- `dotnet-runtime-*`: provide `hostfxr`s for now on
- `dotnet-sdk-{3.1, 6, 7}`: now provide target packs and apphost packs that are useful for Visual Studio/third-party analyzers & debuggers

Version changes:
- `dotnet-runtime-7` & `aspnetcore-runtime-7`: update to 7.0.13
- `dotnet-sdk-7`: update to 7.0.403
- `dotnet-runtime-6` & `aspnetcore-runtime-6`: update to 6.0.24
- `dotnet-sdk-6`: update to 6.0.416
- `*-2.1` & `*-3.1`: mark as EOL

###### Type(s)

- [ ] bugfix
- [x] enhancement
- [ ] security fix

###### Tested on

macOS 13.6 22G120 arm64
Command Line Tools 15.0.0.0.1.1694021235

###### Verification 
Have you

- [x] followed our [Commit Message Guidelines](https://trac.macports.org/wiki/CommitMessages)?
- [x] squashed and [minimized your commits](https://guide.macports.org/#project.github)?
- [x] checked that there aren't other open [pull requests](https://github.com/macports/macports-ports/pulls) for the same change?
- [ ] referenced existing tickets on [Trac](https://trac.macports.org/wiki/Tickets) with full URL? 
- [x] checked your Portfile with `port lint --nitpick`?
- [ ] tried existing tests with `sudo port test`?
- [x] tried a full install with `sudo port -vst install`?
- [x] tested basic functionality of all binary files?
- [ ] checked that the Portfile's most important [variants](https://trac.macports.org/wiki/Variants) haven't been broken?

### CODE REVIEW

**Review by judaew — APPROVED** [review]
LGTM

**Comment by judaew** [comment]
@tsabirgaliev Can you also look at these changes?

File: dotnet/aspnetcore-runtime-2.1/Portfile [hunk 1]
```diff
@@ -34,15 +37,19 @@ extract.mkdir       yes
 
 use_configure       no
 
-depends_run         port:dotnet-cli
+depends_run         port:dotnet-cli \
+                    port:dotnet-runtime-2.1
```
Comment (tsabirgaliev) [inline dotnet/aspnetcore-runtime-2.1/Portfile:41]: @silverbzez I initially thought about this design, but feel that it requires synchronizing aspnetcore-runtime/dotnet-runtime versions. My lack of expertise in both dotnet and macports didn't allow me to come up with a better solution than to have aspnetcore/dotnet as independent ports. BTW, that was what I also observed when installing them using vendor packages.

Comment (silverbzez) [reply dotnet/aspnetcore-runtime-2.1/Portfile:41]: From what I have tested, `dotnet-runtime` is a must-required dependency for `aspnetcore-runtime` as `dotnet-runtime` provides standard libraries like io, data structures, reflections, etc. that are definitely needed for a complicated aspnet project. They can't be independent ports anyway.
However, both versions can be unsynced as long as they are of the same major version. Most of the updates are just sercurity fixes that are guaranteed ABI compatible.

**Review by tsabirgaliev — APPROVED** [review]
LGTM

**Comment by silverbzez** [comment]
@judaew I think we are good to merge now?

File: dotnet/dotnet-cli/Portfile [hunk 2]
```diff
@@ -31,15 +52,15 @@ master_sites        https://dotnetcli.azureedge.net/dotnet/Runtime/${version}/
...
+                            size    30866950
     }
     arm64 {
         distname            dotnet-runtime-${version}-osx-arm64
-        checksums           rmd160  486342ae4505bb501c40bd929668b91ff162b0e2 \
-                            sha256  cbd2cd5825e6d55155d0293e2bf0c08fd0a206f4c5a939690b087c76d18b6d50 \
-                            size    29178452
+        checksums           rmd160  8e9024427b4a7426060afd8cdb36ee9db04656f0 \
+                            sha256  43a129205152389a201d921fcf14565a316bc2a6a271354a08879ad5e4f9d990 \
+                            size    29186244
```
Comment (tsabirgaliev) [inline dotnet/dotnet-cli/Portfile:63]: @silverbzez probably a copy-paste error here, see: https://trac.macports.org/ticket/68580

Comment (silverbzez) [reply dotnet/dotnet-cli/Portfile:63]: Oops my bad, will fix it right away.

**Pull request merged**